### PR TITLE
Improve pkgdown articles; add a new article on filling layouts

### PR DESCRIPTION
**Pull request by cpsievert**

### CODE REVIEW

File: R/card.R [hunk 1]
```diff
@@ -163,6 +165,13 @@ card_body_fill <- function(..., gap = NULL, max_height = NULL, max_height_full_s
   )
 }
 
+#' @export
+#' @keywords internal
+card_body_fill <- function(...) {
+  .Deprecated("card_body_fillable")
```
Comment (cpsievert) [inline R/card.R:171]: Use the rlang way if deprecating?

File: vignettes/sidebars.Rmd [hunk 2]
```diff
@@ -1,38 +1,50 @@
 ---
 title: "Sidebars"
+output:
+  html_document:
+    theme:
+      version: 5
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:6]: Is this needed? Or just for your own development? FWIW, I use something like `nodemon --exec ./build.sh --watch vignettes --ext Rmd` (with `build.sh` containing something like `R -e "pkgdown::build_article('sidebars')"`

(If you have a better "hot reloading" setup feel free to let me know :))

```suggestion
```

File: vignettes/cards.Rmd [hunk 3]
```diff
@@ -6,20 +6,13 @@ resource_files:
...
-One major feature that bslib adds to Bootstrap cards is the ability to expand the card to a [full screen view](#responsive-sizing). Often this feature wants [output that resizes itself to fit its card container](#responsive-sizing). To do this as advertised, make sure you have the latest version of shiny and htmlwidgets:
+## Setup code
 
-```r
-install.packages("shiny")
-install.packages("htmlwidgets")
-```
-
-Since this article is statically hosted (i.e., not powered by Shiny), it uses statically rendered [htmlwidgets](http://www.htmlwidgets.org/) like `{plotly}` and `{leaflet}` (but don't worry, `card()`s [work in Shiny equally as well](#dynamic-rendering-shiny)). Here's some code to create those widgets:
+Since this article is statically hosted (i.e., not powered by Shiny), it uses statically rendered [htmlwidgets](http://www.htmlwidgets.org/) like `{plotly}` and `{leaflet}` (but don't worry, `card()`s [work in Shiny equally as well](#shiny)). Here's some code to create those widgets:
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:11]:
```suggestion
Since this article is statically hosted and not powered by Shiny, it uses statically rendered [htmlwidgets](http://www.htmlwidgets.org/) like `{plotly}` and `{leaflet}`. Don't worry, `card()`s [work in Shiny equally as well](#shiny). Here's some code to create those widgets:
```

File: vignettes/cards.Rmd [hunk 4]
```diff
@@ -113,98 +106,85 @@ card(
...
-  ),
-  card_body(
-    height = 150, 
-    lorem_ipsum_dolor_sit_amet
-  )
-)
-```
-:::
-:::
+By default, a `card()`'s size grows to accommodate the size of it's contents. Thus, if a `card_body()` contains a large amount of text, table(s), etc., you may want to specify a `height`. Although, when laying out [multiple cards](#multiple-cards), you probably shouldn't specify a `height` on the card-level (instead, the height should probably come from a containing element, like `layout_column_wrap()`).
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:119]:
```suggestion
By default, a `card()`'s size grows to accommodate the size of it's contents. Thus, if a `card_body()` contains a large amount of text, tables, or other elements, you may want to specify a `height`. That said, when laying out [multiple cards](#multiple-cards), you probably shouldn't specify a `height` on the card-level. Instead, the height should probably come from a containing element, like `layout_column_wrap()`.
```

File: vignettes/cards.Rmd [hunk 5]
```diff
@@ -113,98 +106,85 @@ card(
...
+  full_screen = TRUE,
   card_header(
     "A long, scrolling, description"
   ),
-  card_body(
-    fill = TRUE,
-    lorem_ipsum_dolor_sit_amet
+  lapply(
+    lorem::ipsum(paragraphs = 3, sentences = c(5, 5, 5)),
+    tags$p
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:134]:
```suggestion
  lorem::ipsum(paragraphs = 3, sentences = 5)
```

`sentences` of length 1 are recycled and `lorem::ipsum()` has an `as.tags()` method now :)

File: vignettes/sidebars.Rmd [hunk 6]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
+    navs_bar(
+      collapsible = FALSE,
+      fillable = TRUE,
+      sidebar = sidebar("Shared sidebar", width = "33%", class = "bg-light"),
+      nav("Page 1", "Page 1 content"),
+      nav("Page 2", "Page 2 content")
+    ),
+    tags$script(HTML("$(document).ready(function() {
+      var demo_nav = $('#page_navbar_example_card nav.navbar');
+      setTimeout(function(){ demo_nav.headroom('destroy') }, 100);
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:150]: Why is this needed? If it is, can we add a comment about why it is? (and also, maybe, check that headroom is on the page before using it since it's a pkgdown thing?)

File: vignettes/cards.Rmd [hunk 7]
```diff
@@ -113,98 +106,85 @@ card(
...
 
-
-## Responsive sizing
+## Filling contents
 
 <div class="row">
 <div class="col-md-6">
 
-Unlike `card_body()`, `card_body_fill()` encourages its children to grow and shrink vertically as needed in response to its `card()`'s height. Responsive sizing is particularly useful for `card(full_screen = TRUE, ...)`, which adds an icon (displayed on hover) to expand the `card()` to a full screen view.
+When a `card()` (or one of it's parent elements) is opinionated about it's height (i.e., `height` is specified and/or `full_screen = TRUE`), it's often convenient to allow re-sizable contents (e.g., plots, tables, etc) to grow and shrink as needed to fit inside the card body (i.e., no scrolling). This can be done by first using `card_body_fillable()` to create a fillable card body container and then placing "fill items" inside of the fillable container.
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:147]:
```suggestion
When a `card()` (or one of it's parent elements) is opinionated about it's height -- i.e., `height` is specified and/or `full_screen = TRUE` -- it's often convenient to allow re-sizable contents (e.g., plots, tables, etc) to grow and shrink as needed to fit inside the card body so the user doesn't need to scroll inside the card. This can be done by first using `card_body_fillable()` to create a fillable card body container and then placing "fill items" inside of the fillable container.
```

File: vignettes/cards.Rmd [hunk 8]
```diff
@@ -244,42 +227,25 @@ card(
...
   card_header("Plot with long description"),
-  card_body_fill(plotly_widget),
+  card_body_fillable(plotly_widget),
   card_body(
     height = "30%", 
-    lorem_ipsum_dolor_sit_amet
+    lapply(
+      lorem::ipsum(paragraphs = 3, sentences = c(5, 5, 5)),
+      tags$p
+    )
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:248]:
```suggestion
    lorem::ipsum(paragraphs = 3, sentences = 5)
```

File: vignettes/cards.Rmd [hunk 9]
```diff
@@ -290,7 +256,7 @@ card(
 
 ::: row
 ::: col-md-6
-Both `card_body()` and `card_body_fill()` include padding between their contents and the `card()` container by default. In either case, you can override those defaults with Bootstrap's [spacing utility classes](https://getbootstrap.com/docs/5.2/utilities/spacing/), like `"p-0"` to remove the padding altogether. This is especially useful if
+Both `card_body()` and `card_body_fillable()` include padding between their contents and the `card()` container by default. In either case, you can override those defaults with Bootstrap's [spacing utility classes](https://getbootstrap.com/docs/5.2/utilities/spacing/), like `"p-0"` to remove the padding altogether. This is especially useful if
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:259]:
```suggestion
Both `card_body()` and `card_body_fillable()` include padding between their contents and the `card()` container by default. In either case, you can override those defaults with Bootstrap's [spacing utility classes](https://getbootstrap.com/docs/5.2/utilities/spacing/), like applying `"p-0"` to remove the padding altogether. This is especially useful if
```

File: vignettes/sidebars.Rmd [hunk 10]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
 
-### Sidebar layouts
+::: lead
+Sidebar layouts in web interfaces allow your users to easily access filters, settings and other inputs alongside the interactive features they control.
+In this article, you'll learn how to create sidebar layouts with bslib.
+:::
+
+## Introduction
+
+To create a sidebar layout, provide a `sidebar()` to the `sidebar` argument of one of the following functions.
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:119]: You're probably aware of this, and I'm not sure how best to address it, but I find this sentence a little bit awkward (since you have to keep reading through multiple paragraphs to see what functions you're referencing). Maybe you could mention/summarize them in a sentence or two?

File: vignettes/cards.Rmd [hunk 11]
```diff
@@ -316,7 +282,7 @@ card(
 
 ::: row
 ::: col-md-6
-Utility classes are really useful since they not only also help with spacing and alignment of stuff _within_ a `card_body()` (or `card_body_fill()`), but more generally enable easy customization of colors, fonts, and more.
+Utility classes are really useful since they not only also help with spacing and alignment of stuff _within_ a `card_body()` (or `card_body_fillable()`), but more generally enable easy customization of colors, fonts, and more.
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:285]:
```suggestion
Not only are utility classes useful to help with spacing and alignment of stuff _within_ a `card_body()` (or `card_body_fillable()`), but they also enable easy customization of [colors](https://getbootstrap.com/docs/5.3/utilities/colors/), [fonts](https://getbootstrap.com/docs/5.3/utilities/text/), and [more](https://getbootstrap.com/docs/5.3/utilities/).
```

File: vignettes/cards.Rmd [hunk 12]
```diff
@@ -440,7 +406,7 @@ card(
 
 ::: row
 ::: col-md-6
-`navs_tab_card()` (as well as `navs_pill_card()`) makes it easy to create cards with multiple tabs (or pills). These functions have the same `full_screen` capabilities as normal `card()`s as well some other options like `title` (since there is no natural place for a `card_header()` to be used). Note that, similar to `card()`, the children of each `nav()` panel will be implicitly wrapped in a `card_body()` call, so use `card_body_fill()` where appropriate to get [responsive sizing](#responsive-sizing).
+`navs_tab_card()` (as well as `navs_pill_card()`) makes it easy to create cards with multiple tabs (or pills). These functions have the same `full_screen` capabilities as normal `card()`s as well some other options like `title` (since there is no natural place for a `card_header()` to be used). Note that, similar to `card()`, the children of each `nav()` panel will be implicitly wrapped in a `card_body()` call, so use `card_body_fillable()` where appropriate to get [filling contents](#filling-contents).
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:409]:
```suggestion
`navs_tab_card()` and `navs_pill_card()` make it easy to create cards with multiple tabs or pills. These functions have the same `full_screen` capabilities as normal `card()`s as well some other options like `title` (since there is no natural place for a `card_header()` to be used). Note that, similar to `card()`, the children of each `nav()` panel will be implicitly wrapped in a `card_body()` call, so use `card_body_fillable()` where appropriate to get [filling contents](#filling-contents).
```

File: vignettes/cards.Rmd [hunk 13]
```diff
@@ -489,7 +455,10 @@ card(
   ),
   card_body(
     height = "30%", class = "pt-0",
-    lorem_ipsum_dolor_sit_amet
+    lapply(
+      lorem::ipsum(paragraphs = 3, sentences = c(5, 5, 5)),
+      tags$p
+    )
```
Comment (gadenbuie) [inline vignettes/cards.Rmd:461]:
```suggestion
    lorem::ipsum(paragraphs = 3, sentences = 5)
```

File: vignettes/sidebars.Rmd [hunk 14]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
+describe_layout_function(
+  "layout_sidebar()",
+  "A flexible, local sidebar layout.",
+  card_body_fillable(
+    class = "p-2",
+    height = 200,
+    layout_sidebar(
+      sidebar = sidebar("Local sidebar", width = "33%", class = "bg-light"),
+      "Content with a sidebar"
+    )
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:199]: The border/border-radius is removed/changes since it's within a card, but you can add them back these utility classes

```suggestion
    layout_sidebar(class = "border rounded-top",
      sidebar = sidebar("Local sidebar", width = "33%", class = "bg-light"),
      "Content with a sidebar"
    )
```

File: vignettes/sidebars.Rmd [hunk 15]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
+::: {.callout .callout-warning}
+<h3 data-toc-skip>Prefer local sidebars</h3>
 
-Although a [global](#global-sidebar) sidebar approach has been popularized by things like `{shinydashboard}`, they're not always ideal for creating a good user experience (UX). As a general UX rule, sidebar controls should be grouped as closely as possible with the contents that it effects. So, if you have multiple groups of outputs you want to show on the same page, you should probably opt for a [local](#local-sidebar) instead of a global sidebar.
+Although [shared sidebars](#shared-sidebar) have been popularized by packages like [shinydashboard](http://rstudio.github.io/shinydashboard/), they're not always ideal for creating a good user experience.
 
+As a general rule, group sidebar controls as closely as possible with the contents they affect.
+Use `layout_sidebar()` to group inputs in a [local sidebar](#local-sidebar) adjacent to the output they control.
+Use global sidebars sparingly and only when the inputs in them are use across many outputs on your page.
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:213]:
```suggestion
Use global sidebars sparingly and only when their controls effect all the outputs on the page.
```

File: _pkgdown.yml [hunk 16]
```diff
@@ -131,3 +138,10 @@ reference:
     - theme_bootswatch
     - bootswatch_themes
     - version_default
+
+
+  redirects:
+    - ["articles/cards.html#fixed-sizing", "articles/cards.html#scrolling-contents"]
```
Comment (gadenbuie) [inline _pkgdown.yml:144]: Unfortunately I don't think pkgdown's redirection system works for within-page redirects. IIRC it's a `<meta http-equiv="refresh">` kind of redirect, so it can only redirect from one page to another.

File: vignettes/sidebars.Rmd [hunk 17]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
 
-### Setup code
+Note that while we talk about shared and local sidebars in this article, those are properties of the interface design.
+Controls placed in a sidebar aren't inherently limited affecting only the outputs next to the sidebar.
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:216]:
```suggestion
Controls placed in a sidebar aren't inherently limited to affecting only the outputs next to the sidebar.
```

File: vignettes/sidebars.Rmd [hunk 18]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
+::::
+
+:::: {.row}
+::: {.col-sm-12 .col-lg-6}
+### Local Sidebar {#intro-local-sidebar}
+
+Use `layout_sidebar()` whenever you want a **[local sidebar](#local-sidebar)** adjacent to a content panel.
+`layout_sidebar()` creates a bordered area by default, making it useful anywhere in your interface.
+
+It also fits nicely into [card components](#cards) and [nav pages or tabs](#navbar-pages) allowing you to add a sidebar to individual pages or tabs.
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:185]:
```suggestion
It also fits nicely into [card components](#cards) and [nav pages or tabs](#navbar-pages) allowing you to add sidebar(s) within individual pages or tabs.
```

File: vignettes/sidebars.Rmd (hunk 6 above)
Comment (gadenbuie) [reply vignettes/sidebars.Rmd:150]: Oh right, yeah it's pkgdown thing where pkgdown enables headroom for anything that matches `nav.navbar`. That probably should be made a little more specific in pkgdown, but I'll add a comment here and check `.headroom` before calling `'destroy'`.

File: vignettes/sidebars.Rmd [hunk 19]
```diff
@@ -58,215 +87,398 @@ iframe, .bslib-page-fill {
...
+
+## Introduction
+
+To create a sidebar layout, provide a `sidebar()` to the `sidebar` argument of one of the following functions.
+
+:::: {.row}
+::: {.col-sm-12 .col-lg-6}
+### Shared Sidebar {#intro-shared-sidebar}
+
+When `sidebar` is an argument to a function containing multiple pages or panels, it creates a **[shared sidebar](#shared-sidebar)**.
```
Comment (cpsievert) [inline vignettes/sidebars.Rmd:125]: Since you use tabs below, better to be consistent

```suggestion
When `sidebar` is an argument to a function containing multiple pages or tabs, it creates a **[shared sidebar](#shared-sidebar)**.
```

File: vignettes/sidebars.Rmd (hunk 10 above)
Comment (gadenbuie) [reply vignettes/sidebars.Rmd:119]: Hmm, maybe something like this?

```suggestion
To create a sidebar layout, provide a `sidebar()` to the `sidebar` argument to `layout_sidebar()` or any of the `navs_` or `page_` functions that support sidebars.
```

**Comment by cpsievert** [comment]
Closing this since I decided to break up this PR into several, more digestable PRs 

- #534, for more immediate code changes we know we want to take on
- #536, for the more speculative `is_fill()`/`as_fill()`
- #537, for the pkgdown improvements
